sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -400,30 +400,42 @@ abstract class CudfFirstLastBase extends CudfAggregate {
   }
 }
 
-class CudfFirstIncludeNulls(override val dataType: DataType) extends CudfFirstLastBase {
+class CudfFirstIncludeNulls(override val dataType: DataType) extends CudfNthBase {
   override val includeNulls: NullPolicy = NullPolicy.INCLUDE
   override val offset: Int = 0
   override lazy val name: String = "CudfFirstIncludeNulls"
 }
 
-class CudfFirstExcludeNulls(override val dataType: DataType) extends CudfFirstLastBase {
+class CudfFirstExcludeNulls(override val dataType: DataType) extends CudfNthBase {
   override val includeNulls: NullPolicy = NullPolicy.EXCLUDE
   override val offset: Int = 0
   override lazy val name: String = "CudfFirstExcludeNulls"
 }
 
-class CudfLastIncludeNulls(override val dataType: DataType) extends CudfFirstLastBase {
+class CudfLastIncludeNulls(override val dataType: DataType) extends CudfNthBase {
   override val includeNulls: NullPolicy = NullPolicy.INCLUDE
   override val offset: Int = -1
   override lazy val name: String = "CudfLastIncludeNulls"
 }
 
-class CudfLastExcludeNulls(override val dataType: DataType) extends CudfFirstLastBase {
+class CudfLastExcludeNulls(override val dataType: DataType) extends CudfNthBase {
   override val includeNulls: NullPolicy = NullPolicy.EXCLUDE
   override val offset: Int = -1
   override lazy val name: String = "CudfLastExcludeNulls"
 }
 
+class CudfNthIncludeNulls(override val dataType: DataType,
```

Comment:
Silly question: Why do we need to permute this class through 6 different combinations? Could we not just use a single class, parametrized on `N` and `includeNulls`?

For instance, couldn't `GpuNthValue#commonExpressions` be defined as follows?
```scala
  private lazy val commonExpressions: Seq[CudfAggregate] = 
￼    Seq(new CudfNth(cudfNth.dataType, offsetVal, isIncludeNulls),
￼        new CudfNth(valueSet.dataType, offsetVal, isIncludeNulls))
```